tests/unit/nupic/frameworks/opf/common_models/cluster_params_test.py

```diff
@@ -59,7 +59,9 @@ def testModelParams(self):
       self.assertGreaterEqual(encodersDict['c1']['resolution'], 0.001,
                               "Resolution is too low")
 
-
+    # Ensure incorrect tmImplementation throws exception
+    with self.assertRaises(ValueError):
+        getScalarMetricWithTimeOfDayAnomalyParams([0], tmImplementation="")
```

Comment:
You should also have a test that calls it with tm_cpp and make sure it returns the correct JSON file.